src/alire/alire-directories.adb

```diff
@@ -11,12 +11,18 @@ package body Alire.Directories is
    -- Backup_If_Existing --
    ------------------------
 
-   procedure Backup_If_Existing (File : Any_Path) is
+   procedure Backup_If_Existing (File   : Any_Path;
+                                 Base_Dir : Any_Path := "")
+   is
       use Ada.Directories;
+      Dst : constant String := (if Base_Dir /= ""
+                                then Base_Dir / Simple_Name (File) & ".prev"
+                                else File & ".prev");
    begin
       if Exists (File) then
-         Trace.Debug ("Backing up " & File);
-         Copy_File (File, File & ".prev", "mode=overwrite");
+         Trace.Debug ("Backing up " & File
+                      & " with base dir: " & Base_Dir);
+         Copy_File (File, Dst, "mode=overwrite");
```

Comment:
I think it's better to keep the `.prev` or `.back` so the file is not confusing for users.

Reply:
If I'm understanding correctly, that's what's happening; the files are copias as before with the extra `.prev` but put inside `alire`.

Reply:
My bad, the `.prev` is added above.

Sorry for the noise.